### PR TITLE
ICU-22922 ICU BRS 77: front-load update version to 77.0.1

### PR DESCRIPTION
**Pull request by mihnita**
#### Checklist
- [x] Required: Issue filed: ICU-22922
- [x] Required: The PR title must be prefixed with a JIRA Issue number. Example: "ICU-1234 Fix xyz"
- [x] Required: The PR description must include the link to the Jira Issue, for example by completing the URL in the first checklist item
- [x] Required: Each commit message must be prefixed with a JIRA Issue number. Example: "ICU-1234 Fix xyz"
- [x] Issue accepted (done by Technical Committee after discussion)
- [ ] Tests included, if applicable
- [ ] API docs and/or User Guide docs changed or added, if applicable

### CODE REVIEW

File: docs/processes/release/tasks/versions.md [hunk 1]
```diff
@@ -53,6 +53,8 @@ need to be correspondingly updated. See below for more files to be updated and s
 [icu4c/source/data/misc/icuver.txt](https://github.com/unicode-org/icu/blob/main/icu4c/source/data/misc/icuver.txt)
 needs to be updated with the correct version number for ICU and its data.
 
+### CLDR-to-ICU onversion
```
Comment (markusicu) [inline docs/processes/release/tasks/versions.md:56]:
```suggestion
### CLDR-to-ICU Conversion
```

File: icu4c/source/configure
Comment (markusicu) [inline icu4c/source/configure:1]: no configure.in changes?

**Review by markusicu — CHANGES_REQUESTED** [review]
changes lgtm but the ticket is wrong -- it's the release ticket.
please use the rc ticket: ICU-22922

File: icu4c/source/configure
Comment (mihnita) [reply icu4c/source/configure:1]: Not listed here:
https://unicode-org.github.io/icu/processes/release/tasks/versions.html

And I don't see any `configure.in` file.

```sh
$ find . | grep -i configure
./icu4c/source/configure.ac
./icu4c/source/configure
./icu4c/source/runConfigureICU
./icu4c/as_is/os400/convertConfigure.sed
```

Other than that:
* I tested everything (icu4c and icu4j)
* I compared the CL that did the front-load for 76.0.1 and this one
  They are the same, with the only differences in `tools/cldr/cldr-to-icu` (the change from yesterday) and one change missing from the component that was shared with cldr and was removed last quarter. So all accounted for.

**Comment by mihnita** [comment]
> changes lgtm but the ticket is wrong -- it's the release ticket. please use the rc ticket: [ICU-22922](https://unicode-org.atlassian.net/browse/ICU-22922)

DONE, thanks.

**Review by markusicu — COMMENTED** [review]
lgtm tnx

pse squash, and remember to fix the ticket number in the commit message

**Comment by mihnita** [comment]
> lgtm tnx
> 
> pse squash, and remember to fix the ticket number in the commit message

Thank you.
Sqashed, used "ICU-22922" in description

**Review by markusicu — APPROVED** [review]

**Pull request merged**